filament/src/components/RenderableManager.cpp

```diff
@@ -301,10 +304,49 @@ RenderableManager::Builder::Result RenderableManager::Builder::build(Engine& eng
 }
 
 RenderableManager::Builder& RenderableManager::Builder::instances(size_t instanceCount) noexcept {
+    ASSERT_PRECONDITION(mImpl->mInstanceTransforms == nullptr && mImpl->mInstanceBuffer == nullptr,
+            "cannot call instances(size_t) override, instance local transforms already provided");
     mImpl->mInstanceCount = clamp((unsigned int)instanceCount, 1u, 32767u);
     return *this;
 }
 
+RenderableManager::Builder& RenderableManager::Builder::instances(size_t instanceCount,
+        math::mat4f const* localTransforms) noexcept {
```

Comment:
not sure if we should have both APIs. I think I prefer we be opinionated and pick one.

Reply:
The main reason why I think the "manual" instancing case is useful is that it allows way more instances. Maybe we should go ahead and implement the "instanced chopping" we discussed to allow a large number of instances even with the hybrid version.